nexus/tests/integration_tests/scim.rs

```diff
@@ -2185,3 +2185,189 @@ async fn test_scim_list_users_with_groups(cptestctx: &ControlPlaneTestContext) {
+                format!("Bearer oxide-scim-{}", created_token.bearer_token),
+            )
+            .allow_non_dropshot_errors()
+            .raw_body(Some(
+                serde_json::to_string(&serde_json::json!({
+                    "displayName": "group1",
+                    "externalId": "group1@example.com",
+                    "members": [
+                        {"value": users[0].id},
+                        {"value": users[1].id},
+                        {"value": users[2].id},
+                    ],
+                }))
+                .unwrap(),
+            ))
+            .expect_status(Some(StatusCode::CREATED)),
+    )
+    .execute_and_parse_unwrap()
+    .await;
+
+    let group2: scim2_rs::Group = NexusRequest::new(
+        RequestBuilder::new(client, Method::POST, "/scim/v2/Groups")
+            .header(http::header::CONTENT_TYPE, "application/scim+json")
+            .header(
+                http::header::AUTHORIZATION,
+                format!("Bearer oxide-scim-{}", created_token.bearer_token),
+            )
+            .allow_non_dropshot_errors()
+            .raw_body(Some(
+                serde_json::to_string(&serde_json::json!({
+                    "displayName": "group2",
+                    "externalId": "group2@example.com",
+                    "members": [
+                        {"value": users[0].id},
+                        {"value": users[3].id},
+                    ],
+                }))
+                .unwrap(),
+            ))
+            .expect_status(Some(StatusCode::CREATED)),
+    )
+    .execute_and_parse_unwrap()
+    .await;
+
+    let group3: scim2_rs::Group = NexusRequest::new(
+        RequestBuilder::new(client, Method::POST, "/scim/v2/Groups")
+            .header(http::header::CONTENT_TYPE, "application/scim+json")
+            .header(
+                http::header::AUTHORIZATION,
+                format!("Bearer oxide-scim-{}", created_token.bearer_token),
+            )
+            .allow_non_dropshot_errors()
+            .raw_body(Some(
+                serde_json::to_string(&serde_json::json!({
+                    "displayName": "group3",
+                    "externalId": "group3@example.com",
+                }))
+                .unwrap(),
+            ))
+            .expect_status(Some(StatusCode::CREATED)),
+    )
+    .execute_and_parse_unwrap()
+    .await;
+
+    // List all groups and verify members
+    let response: scim2_rs::ListResponse = NexusRequest::new(
+        RequestBuilder::new(client, Method::GET, "/scim/v2/Groups")
+            .header(http::header::CONTENT_TYPE, "application/scim+json")
+            .header(
+                http::header::AUTHORIZATION,
+                format!("Bearer oxide-scim-{}", created_token.bearer_token),
+            )
+            .allow_non_dropshot_errors()
+            .expect_status(Some(StatusCode::OK)),
+    )
+    .execute_and_parse_unwrap()
+    .await;
```

Comment:
I'm going to combine this with the test from #9325 since the setup is identical.